crates/fuel-core/src/query/balance.rs

```diff
@@ -28,29 +29,28 @@ use std::{
 pub mod asset_query;
 
 impl ReadView {
-    pub fn balance(
+    pub async fn balance(
         &self,
         owner: Address,
         asset_id: AssetId,
         base_asset_id: AssetId,
     ) -> StorageResult<AddressBalance> {
         let amount = AssetQuery::new(
             &owner,
-            &AssetSpendTarget::new(asset_id, u64::MAX, usize::MAX),
+            &AssetSpendTarget::new(asset_id, u64::MAX, u16::MAX),
```

Comment:
there's quite a bit of difference between usize::MAX (18446744073709551615) and u16::MAX (65535)
nice optimization